care/emr/api/viewsets/diagnostic_report.py

```diff
@@ -0,0 +1,168 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from care.emr.api.viewsets.base import EMRModelViewSet
+from care.emr.models.diagnostic_report import DiagnosticReport
+from care.emr.models.observation import Observation
+from care.emr.resources.diagnostic_report.spec import (
+    DiagnosticReportCreateSpec,
+    DiagnosticReportListSpec,
+    DiagnosticReportObservationRequest,
+    DiagnosticReportRetrieveSpec,
+    DiagnosticReportReviewRequest,
+    DiagnosticReportUpdateSpec,
+    DiagnosticReportVerifyRequest,
+    StatusChoices,
+)
+from care.emr.resources.observation.spec import Performer, PerformerType
+
+
+class PhaseChoices(models.TextChoices):
+    IN_PROCESS = "in_process", "In Process"
+    VERIFICATION_REQUIRED = "verification_required", "Verification Required"
+    REVIEW_REQUIRED = "review_required", "Review Required"
+    REVIEWED = "reviewed", "Reviewed"
+
+
+class DiagnosticReportFilters(FilterSet):
+    phase = ChoiceFilter(choices=PhaseChoices.choices, method="filter_phase")
+    status = ChoiceFilter(
+        choices=[(status.value, status.value) for status in StatusChoices]
+    )
+    specimen = UUIDFilter(field_name="specimen__external_id")
+    based_on = UUIDFilter(field_name="based_on__external_id")
+
+    ordering = OrderingFilter(
+        fields=(
+            ("created_date", "created_date"),
+            ("modified_date", "modified_date"),
+        )
+    )
+
+    def filter_phase(self, queryset, name, value):
+        return queryset.annotate(
+            phase=Case(
+                When(status=StatusChoices.final, then=Value("reviewed")),
+                When(status=StatusChoices.preliminary, then=Value("review_required")),
+                When(status=StatusChoices.partial, then=Value("verification_required")),
+                default=Value("in_process"),
+                output_field=CharField(),
+            )
+        ).filter(phase=value)
+
+
+@extend_schema_view(
+    create=extend_schema(request=DiagnosticReportCreateSpec),
+    update=extend_schema(request=DiagnosticReportUpdateSpec),
+    list=extend_schema(request=DiagnosticReportListSpec),
+    retrieve=extend_schema(request=DiagnosticReportRetrieveSpec),
+)
+class DiagnosticReportViewSet(EMRModelViewSet):
+    database_model = DiagnosticReport
+    pydantic_model = DiagnosticReportCreateSpec
+    pydantic_update_model = DiagnosticReportUpdateSpec
+    pydantic_read_model = DiagnosticReportListSpec
+    pydantic_retrieve_model = DiagnosticReportRetrieveSpec
+    filter_backends = [DjangoFilterBackend]
+    filterset_class = DiagnosticReportFilters
+
+    def clean_create_data(self, request, *args, **kwargs):
+        clean_data = super().clean_create_data(request, *args, **kwargs)
+
+        clean_data["performer"] = self.request.user.external_id
```

Comment:
Set the value on perform_create and remove it from the spec for create and update